compiler/rustc_codegen_llvm/src/llvm/mod.rs

```diff
@@ -178,10 +178,10 @@ pub fn SetFunctionCallConv(fn_: &Value, cc: CallConv) {
 // function.
 // For more details on COMDAT sections see e.g., https://www.airs.com/blog/archives/52
 pub fn SetUniqueComdat(llmod: &Module, val: &Value) {
-    unsafe {
-        let name = get_value_name(val);
-        LLVMRustSetComdat(llmod, val, name.as_ptr().cast(), name.len());
-    }
+    let name_buf = get_value_name(val).to_vec();
+    let name =
+        CString::from_vec_with_nul(name_buf).or_else(|buf| CString::new(buf.into_bytes())).unwrap();
+    set_comdat(llmod, val, &name);
```

Comment:
Oh, looks like we need the supports-comdat check here too.

Reply:
Isn't this because `!self.is_like_aix && !self.is_like_osx` should be `!self.is_like_aix || !self.is_like_osx`?

Reply:
I mean it's really a sign that "setting a comdat" needs to be a function on the CodegenCx, imo.

Reply:
Yeah, that makes sense. Happy to leave that for future work.